File: LuaScripts/UI/HighScoreUI.lua
local HighScoreUI = LuaUIManager.Register(LuaUI, "HighScoreUI", UIType.Normal, UIResType.Resorces, UICloseType.Hide, true);

--function HighScoreUI:Ctor(name, ui)

--    self.super.Ctor(self, name, ui);

--end

function HighScoreUI:OnAwake()

    Debug.Log("Lua HighScoreUI OnAwake" .. self.name)
    local BackButton = self.gameObject:FindGameObject("BackButton")
    self.super.RegisterListener(self, BackButton, self.OnClickExitBtn)
    
end

function HighScoreUI:OnClickExitBtn(eventData)
    UIManager.instance:Pop()
end

function HighScoreUI:OnStart(args)
    Debug.Log("Lua HighScoreUI OnStart")
end

function HighScoreUI:OnEnable()
    Debug.Log("Lua HighScoreUI OnEnable")
end

function HighScoreUI:OnDisable()
    Debug.Log("Lua HighScoreUI OnDisable")
end

function HighScoreUI:OnDestroy()
    Debug.Log("Lua HighScoreUI OnDestroy")
end

function HighScoreUI:OnNotifiy(evt, args)
    
end

File: Assets/UIDemo/Scripts/HighScoreUI.cs
using UnityEngine;
using UnityEngine.UI;
using System.Collections;
using System.Collections.Generic;
using System.Threading.Tasks;
using System;
using UIFramework;
using UnityEngine.EventSystems;

//[UIAttribute("HighScoreUI", UIType.Normal, UIResType.Resorces, UICloseType.Hide, true)]
public class HighScoreUI : UIMonoProxy
{
    public override void OnAwake()
    {
        Debug.Log("HighScoreUI OnAwake");
        RegisterListener("BackButton", OnClickExitBtn);
    }

    void OnClickExitBtn(PointerEventData eventData)
    {
        UIManager.Instance.Pop();
    }

    public override void OnStart(params object[] args)
    {
        Debug.Log("HighScoreUI OnStart");
    }

    public override void OnEnable()
    {
        Debug.Log("HighScoreUI OnEnable");
    }

    public override void OnDisable()
    {
        Debug.Log("HighScoreUI OnDisable");
    }

    public override void OnDestroy(bool delete)
    {
        Debug.Log("HighScoreUI OnDestroy");
    }
}